applauncher/004-three-by-three.qml

```diff
@@ -45,68 +46,15 @@ Item {
     property bool dragStop: false
     property string appTitle: ""
     property real clickY: 0
+    property int numColumns: 3
+    
+    Component {
+        id: gridDelegate
 
-    GridView {
-        id: appsView
-
-        flow: GridView.FlowLeftToRight
-        snapMode: GridView.SnapToRow
-        anchors {
-            fill: parent
-            leftMargin: parent.width * .026
-            rightMargin: parent.width * .026
-        }
-        clip: true
-        cellHeight: appsView.height / 3.15
-        cellWidth: appsView.width / 3
-        preferredHighlightBegin: height / 3.15 - currentItem.height / 3.15
-        preferredHighlightEnd: height / 3.15 + currentItem.height / 3.15
-        highlightRangeMode: ListView.StrictlyEnforceRange
```

Comment:
Somehow I have the feeling that these fields were trying to solve something else than the footer/header problem you've solved, yet, I can't quite put my finger on what that would be.

@MagneFire it seems that you wrote this code, maybe you remember what you were trying to achieve here ?

Reply:
Originally, this logic was used to ensure an equal amount of spacing on the top and bottom of the `GridView`. In this launcher it was used to shift the grid a bit further to the top resulting in the weird bug where too much padding is available at the bottom.
Removing this and adding header and footer components is the right way to go here.

Reply:
Thanks for the confirmation.  I read about these fields and came to that conclusion, but it's nice to get feedback.